group/views.py

```diff
@@ -20,3 +23,100 @@ def create_group(request):
             return Response(serializer.data, status=status.HTTP_201_CREATED)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
     return None
+
+
+@api_view(['PATCH'])
+@permission_classes([IsAuthenticated])
+def add_users(request):
+    """Add user(s) to a group"""
+    if request.method == 'PATCH':
+        data = request.data.copy()
+        group_id = request.data.get("group_id")
+
+        data.pop('creator', None)
+        data.pop('name', None)
+        try:
+            group = Group.objects.get(id=group_id)
+            if request.user != group.creator:
+                return Response({
+                                "error": "Only group creator can add members"},
+                                status=status.HTTP_403_FORBIDDEN)
+            errors = []
+            for member in group.members.all():
+                if member.id not in data['members']:
+                    data['members'].append(member.id)
+            for member in data['members']:
+                try:
+                    user = User.objects.get(id=member)
+                    if user.id not in group.members.all()\
+                            .values_list('id', flat=True):
+                        group.members.add(user)
+                except User.DoesNotExist:
+                    errors.append(f"User with ID {member} is not found")
+            serializer = GroupSerializer(group, data=data, partial=True)
+            if serializer.is_valid():
+                serializer.save()
+                return Response(serializer.data, status=status.HTTP_200_OK)
+            if errors:
+                return Response({"error": errors},
+                                status=status.HTTP_404_NOT_FOUND)
+            return Response(serializer.errors,
+                            status=status.HTTP_400_BAD_REQUEST)
+        except Group.DoesNotExist:
+            return Response({"error": "Group not found"},
+                            status=status.HTTP_404_NOT_FOUND)
+    return None
+
+
+@api_view(['DELETE'])
+@permission_classes([IsAuthenticated])
+def delete_group(request):
+    """Delete a group"""
+    if request.method == 'DELETE':
+        group_id = request.data.get("group_id")
+        try:
+            group = Group.objects.get(id=group_id)
+            if request.user != group.creator:
```

Comment:
Read about `permission_classes`... DRT has them and they should be enough for this. Read about the `@transaction.atomic` decorator as well. Do let me know if you need more guidance on these.